javascript/app.js

```diff
@@ -76,54 +92,201 @@ FinancialFreedom.controller('HeaderController', ['$scope', '$location',  functio
         ];
 
         return non_visible_pages.indexOf($location.path() ) == -1;
+    };
 
+    $scope.openLoginModal = function(size, showSignUp) {
+
+        var modalInstance = $modal.open({
+            templateUrl: 'partials/login_modal.html',
+            controller: 'LoginModalInstanceCtrl',
+            size: size,
+            backdrop: true,
+            resolve: {
+                showSignUp: function() {
+                    return showSignUp;
+                }
+            }
+        });
     };
-}]);
 
-FinancialFreedom.controller('IncomeInputController', ['$scope', '$location', 'RetirementCalculatorService', function($scope, $location, RetirementCalculatorService) {
-    $scope.income = {};
-    $scope.income.value = RetirementCalculatorService.getMonthlyIncome();
+    $scope.openAccountModal = function(size) {
+
+        var modalInstance = $modal.open({
+            templateUrl: 'partials/account_modal.html',
+            controller: 'AccountModalInstanceCtrl',
+            size: size,
+            backdrop: true,
+        });
+    };

```

Comment:
this same chunk of code for opening the modal seems to be in a few different controllers, any way to modularize it?